drivers/Kconfig

```diff
@@ -97,4 +97,6 @@ source "drivers/eeprom/Kconfig"
 
 source "drivers/peci/Kconfig"
 
+source "drivers/battery/Kconfig"
```

Comment:
driver infrastructure and Kconfig should come with a driver.

Reply:
This is implemented based on the discussions in https://github.com/zephyrproject-rtos/zephyr/issues/5554. The actual driver implementation is defined out of tree. Not sure how to open that up for review.